### PR TITLE
issue #81

### PR DESCRIPTION
**Pull request by ThumarHarsh**
…ionality :On clicking on the volunteer name (appearing anywhere throughout the app), the volunteer's profile should open in a new tab.

# Related Issue
Fixes: #81

# Prosposed Changes
- add link onvolunteers  profile click in anywhere in application

### CODE REVIEW

File: templates/volunteers/attendance.html [hunk 1]
```diff
@@ -33,7 +33,7 @@ <h3 class="text-center">Volunteers Attendance</h3>
                         {% for vol_att in today_vol_att %}
                         <tr>
                             <td>{{ vol_att.volun.roll_no }}</td>
-                            <td>{{ vol_att.volun.profile.get_full_name }}</td>
+                            <td><a href="{% url 'volunteers:profile' vol_att.volun.profile.user_id%}">{{ vol_att.volun.profile.get_full_name }}</a></td>
```
Comment (garg3133) [inline templates/volunteers/attendance.html:36]: Can you format it better? Put each element in a separate line.

File: templates/volunteers/volunteers_list.html [hunk 2]
```diff
@@ -64,7 +64,7 @@ <h4 class="text-center mt-3">Registered Volunteers</h4>
                     let vol_list_table = '<thead><tr><th>Roll No.</th><th>Name</th><th>Section</th></tr></thead><tbody>';
 
                     Object.entries(data).forEach(([key, [roll_no, name, section]]) => {
-                        vol_list_table += '<tr><td>' + roll_no + '</td><td>' + name + '</td><td>' + section + '</td></tr>';
+                        vol_list_table += '<tr><td>' +  roll_no + '</td><td><a href="{% url 'volunteers:profile' profile.user_id%}">' + name + '</a></td><td>' + section + '</td></tr>';
```
Comment (garg3133) [inline templates/volunteers/volunteers_list.html:67]: Remove the extra space before `roll_no`

**Review by garg3133 — COMMENTED** [review]
There's one more page where this needs to be implemented: Dashboard. Go to Admin Portal > Profiles and assign yourself any designation (there's a `desig` field at the end of the page). You'll be able to view Volunteer's attendance then on the dashboard.

File: templates/volunteers/attendance.html (hunk 1 above)
Comment (ThumarHarsh) [reply templates/volunteers/attendance.html:36]: ok, I will do that.

File: templates/volunteers/volunteers_list.html (hunk 2 above)
Comment (ThumarHarsh) [reply templates/volunteers/volunteers_list.html:67]: okk.

**Comment by ThumarHarsh** [comment]
I have formatted two files and also add the functionality of: "On clicking on the volunteer name (appearing anywhere throughout the app), the volunteer's profile should open in a new tab."  added in dashboard.html page. please check it.

**Comment by garg3133** [comment]
@ThumarHarsh Few issues I found with your PR:
* On `volunteers/list` page, all links are directing to the same profile.
* Links are opening in the same tab instead of opening in a new tab.
* On dashboard, `<space>` after the name shouldn't be included in the link.

**Comment by ThumarHarsh** [comment]
> @ThumarHarsh Few issues I found with your PR:
> 
> * On `volunteers/list` page, all links are directing to the same profile.
> * Links are opening in the same tab instead of opening in a new tab.
> * On dashboard, `<space>` after the name shouldn't be included in the link.

**Comment by ThumarHarsh** [comment]
okk,i will try

File: templates/volunteers/volunteers_list.html [hunk 3]
```diff
@@ -57,14 +57,15 @@ <h4 class="text-center mt-3">Registered Volunteers</h4>
...
                 }
-                else{
+                else {
                     let vol_list_table = '<thead><tr><th>Roll No.</th><th>Name</th><th>Section</th></tr></thead><tbody>';
 
-                    Object.entries(data).forEach(([key, [roll_no, name, section]]) => {
-                        vol_list_table += '<tr><td>' + roll_no + '</td><td><a href="{% url 'volunteers:profile' profile.user_id%}">' + name + '</a></td><td>' + section + '</td></tr>';
+                    Object.entries(data).forEach(([key, [user_id, roll_no, name, section]]) => {
+                        const url_mask = "{% url 'volunteers:profile' 14325 %}".replace("14325", user_id);
+                        vol_list_table += '<tr><td>' + roll_no + '</td><td><a href="' + url_mask + '" target =_blank>' + name + '</a></td><td>' + section + '</td></tr>';
```
Comment (garg3133) [inline templates/volunteers/volunteers_list.html:68]: `_blank` should come in double quotes.

File: apps/volunteers/views.py [hunk 4]
```diff
@@ -279,6 +279,6 @@ def ajax_update_schedule(request):
     schedule = Schedule.objects.filter(day=sch_day).order_by('section__section_id')
 
     for sch in schedule:
-        data[sch.section.section_id] = sch.section.name
+        data[isch.section.section_d] = sch.section.name
```
Comment (garg3133) [inline apps/volunteers/views.py:282]: You made this change by mistake, I guess!

File: apps/volunteers/views.py [hunk 5]
```diff
@@ -150,7 +150,7 @@ def ajax_volunteers_list(request):
     # and 'roll_no' to make every key unique.
     for vol_sch in vol_to_show:
         key = str(vol_sch.schedule.section.section_id) + str(vol_sch.volun.roll_no)
-        data[key] = [vol_sch.volun.roll_no, vol_sch.volun.profile.get_full_name, vol_sch.schedule.section.name]
+        data[key] = [vol_sch.volun.profile.user_id,vol_sch.volun.roll_no, vol_sch.volun.profile.get_full_name, vol_sch.schedule.section.name]
```
Comment (garg3133) [inline apps/volunteers/views.py:153]: You should leave a `<space>` after `,`.

File: templates/home/dashboard.html [hunk 6]
```diff
@@ -208,10 +208,11 @@ <h4 class="text-center mt-3">Volunteers Present</h4>
                         <tr class="text-center">
                             <td class="text-monospace font-weight-lighter">{{ vol_att.volun.roll_no }}</td>
                             <td class="text-monospace font-weight-lighter">
-                                <a href="{% url 'volunteers:profile'  vol_att.volun.profile.user_id%}">
-                                   {{ vol_att.volun.profile.get_full_name }}
-                                </a> 
-                            {% if vol_att.extra %}(E){% endif %}</td>
+                                 <a href="{% url 'volunteers:profile' vol_att.volun.profile.user_id%}" target ="_blank">
+                                   {{ vol_att.volun.profile.get_full_name }}</a>
+                                    {% if vol_att.extra %}(E){% endif %}
```
Comment (garg3133) [inline templates/home/dashboard.html:213]: Format it properly like before (ending anchor tag in a separate line).

File: templates/volunteers/attendance.html [hunk 7]
```diff
@@ -34,7 +34,7 @@ <h3 class="text-center">Volunteers Attendance</h3>
                         <tr>
                             <td>{{ vol_att.volun.roll_no }}</td>
                             <td>
-                                <a href="{% url 'volunteers:profile' vol_att.volun.profile.user_id%}">
+                                <a href="{% url 'volunteers:profile' vol_att.volun.profile.user_id%}" target ="_blank">
```
Comment (garg3133) [inline templates/volunteers/attendance.html:37]: Leave a space after `id` and before `%`.

**Review by garg3133 — COMMENTED** [review]
After doing the below changes, can you please squash your commits? Steps to be followed:
* `git rebase -i HEAD~<no. of commits you made in total>`
* Except for first `pick`, replace all `pick` with `squash` in the file opened.
* Close the file after saving it.
* In the new file opened, remove all commit messages and write a new one and then close this file too after saving.
* Force push your changes. (`git push origin +<your-branch-name>`).

File: templates/home/dashboard.html (hunk 6 above)
Comment (ThumarHarsh) [reply templates/home/dashboard.html:213]: @garg3133 if I put the closing anchor tag in a separate line then there is <space> after the name which shouldn't be included in the link.

**Comment by garg3133** [comment]
Merged! :tada:

Thanks, @ThumarHarsh for your contribution 🚀

**Pull request merged**